Sources/Purchasing/StoreKitAbstractions/Storefront.swift

```diff
@@ -69,6 +69,32 @@ public class Storefront: NSObject, StorefrontType {
 
 }
 
+public extension Storefront {
+
+    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, macCatalyst 13.1, *)
+    private static var currentStorefrontType: StorefrontType? {
+        get async {
+            if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *) {
+                let sk2Storefront = await StoreKit.Storefront.current
+                return sk2Storefront.map(SK2Storefront.init)
+            } else {
+                return SKPaymentQueue.default().storefront.map(SK1Storefront.init)
+            }
+        }
+    }
+
+    /// The current App Store storefront for the device.
+    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, macCatalyst 13.1, *)
+    static var currentStorefront: Storefront? {
```

Comment:
I named it `currentStorefront` because `Storefront.current` would be ambiguous. Ideally we'd come up with a different name for our `Storefront` to avoid that but I couldn't think of anything.
We've done that in the past for other types by adding the word "store", like "StoreProduct", but that doesn't work in this case 😛

Reply:
It could be worse, but I'm ok with this.